frame/assets/src/lib.rs

```diff
@@ -179,7 +179,9 @@ decl_module! {
 			ensure!(origin_balance >= amount, "origin account balance must be greater than or equal to the transfer amount");
 
 			Self::deposit_event(RawEvent::Transferred(id, origin, target.clone(), amount));
+			// we check above that this will not overflow
 			<Balances<T>>::insert(origin_account, origin_balance - amount);
+			// sum of all issuances cannot exceed u64, so this will not overflow WARN
```

Comment:
this is not true, as i said in the chat. however, *in the polkadot codebase* `TotalIssuance` will make it above 2**96 and `Balance` is `u128`.